errors/UnauthorizedError.ts

```diff
@@ -0,0 +1,10 @@
+type Reason =
+    | "Not a TokoBapak member"
+    | "Not logged in"
+    | "Requires higher access";
+
+export class UnauthorizedError extends Error {
+    constructor(public readonly reason: Reason) {
+        super(reason)
+    }
+}
```

Comment:
ku prefer pake enum aja sih daripada string union, biar semisal kalo messagenya diubah semua callernya ga perlu diubah

- pake const biar autocomplete tau isi enumnya
- pake `typeof REASON[keyof typeof REASON]` biar di parameter hintnya muncul teks aslinya

```suggestion
export const REASON = {
    not_tokobapak_member: "Not a TokoBapak member",
    not_logged_in: "Not logged in",
    requires_higher_access: "Requires higher access",
} as const;
type Reason = typeof REASON[keyof typeof REASON];

export class UnauthorizedError extends Error {
    constructor(public readonly reason: Reason) {
        super(reason)
    }
}
```